app/controllers/comments_controller.rb

```diff
@@ -9,14 +9,15 @@ def new
 
   def create
 
-    p "here is there"
-    @post = Post.find(params[:post_id])
-    @comment = @post.comments.create(comment_params) do |comment|
-      comment.user_id=current_user[:id]
-    end
-    #debugger
+
+    @post = Post.find_by(id: params[:post_id])
+    @comment = @post.comments.create(comment_params)
```

Comment:
this code will break in case @post is `nil`. please handle that case as well.

Reply:
How can a post be nil if create comment option is only visible on post?

Reply:
But you still need to handle to case. May be in future this code evolve into something else

Reply:
Done